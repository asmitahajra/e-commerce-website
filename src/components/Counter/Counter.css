.counter {
    display: flex
}

button {
    margin: 10px 10px;
}

.inBasket {
    background-color:darkkhaki;
}