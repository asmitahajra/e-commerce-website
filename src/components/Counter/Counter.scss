@import '../../styles/index.scss';

.counter {
    display: flex;
    button {
        margin: 1rem 1rem;
        @extend .button-outline;
    }
    .inBasket {
        // background-color: $primary-backgroud-color;
        background-color:darkkhaki;
    }
    // p {
    //     @include custom-mixin;
    // }
}
