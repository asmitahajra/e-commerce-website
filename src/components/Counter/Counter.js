import React, { useContext } from 'react';
import './Counter.scss';
import PropTypes from 'prop-types';
import { ThemeContext } from '../../ThemeContext';
import { BasketContext } from '../../BasketContext';

const Counter = ({ count, onDecrement, onIncrement }) => {
  const Theme = useContext(ThemeContext);
  const Basket = useContext(BasketContext);

  return (
    <div className="counter">
      <button type="button" onClick={onIncrement} style={{ background: Theme.backgroundColor, color: Theme.color }}>+</button>
      <p className="inBasket" style={{ background: Basket.backgroundColor }}>
        {count}
        {' '}
        in Basket
      </p>
      <button type="button" onClick={onDecrement} style={{ background: Theme.backgroundColor, color: Theme.color }}>-</button>
    </div>
  );
};

Counter.propTypes = {
  count: PropTypes.number.isRequired,
  onDecrement: PropTypes.func.isRequired,
  onIncrement: PropTypes.func.isRequired,
};

export default Counter;
