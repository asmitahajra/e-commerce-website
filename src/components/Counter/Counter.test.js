/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Counter from './Counter';
// import { Theme, ThemeContext } from '../../ThemeContext';

describe(Counter.name, () => {
  const mockProp = {
    count: 0,
    onIncrement: jest.fn(),
    onDecrement: jest.fn(),
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('sanity', () => {
    expect(1 + 1).toBe(2);
  });

  test('should match snapshot', () => {
    const { container } = render(<Counter {...mockProp} />);
    expect(container).toMatchSnapshot();
  });

  test('should display an increment button', () => {
    render(<Counter {...mockProp} />);
    const incrementElement = screen.getByText('+');
    expect(incrementElement.tagName).toBe('BUTTON');
  });

  test('should display an increment button', () => {
    render(<Counter {...mockProp} />);
    const decrementElement = screen.getByText('-');
    expect(decrementElement.tagName).toBe('BUTTON');
  });

  test('should display the current count', () => {
    render(<Counter {...mockProp} />);
    expect(screen.getByText('0 in Basket'));
  });

  test('should fire onIncrement function on clicking +', () => {
    render(<Counter {...mockProp} />);
    const incrementCount = screen.getByText('+');
    fireEvent.click(incrementCount);
    expect(mockProp.onIncrement).toHaveBeenCalledTimes(1);
  });

  test('should fire onDecrement function on clicking -', () => {
    render(<Counter {...mockProp} />);
    const decrementCount = screen.getByText('-');
    fireEvent.click(decrementCount);
    expect(mockProp.onDecrement).toHaveBeenCalledTimes(1);
    // expect(mockProp.onDecrement).not.toHaveBeenCalled();
  });
});
