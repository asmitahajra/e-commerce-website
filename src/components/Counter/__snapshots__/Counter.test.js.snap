// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should match snapshot 1`] = `
<div>
  <div
    class="counter"
  >
    <button
      style="background: white; color: black;"
      type="button"
    >
      +
    </button>
    <p
      class="inBasket"
      style="background: gray;"
    >
      0
       
      in Basket
    </p>
    <button
      style="background: white; color: black;"
      type="button"
    >
      -
    </button>
  </div>
</div>
`;
